roles/openshift_health_checker/test/action_plugin_test.py

```diff
@@ -249,3 +305,38 @@ def test_resolve_checks_failure(names, all_checks, words_in_exception):
         resolve_checks(names, all_checks)
     for word in words_in_exception:
         assert word in str(excinfo.value)
+
+
+@pytest.mark.parametrize('give_output_dir, result, expect_file', [
+    (False, None, False),
+    (True, dict(content="c3BhbQo=", encoding="base64"), True),
+    (True, dict(content="encoding error", encoding="base64"), False),
+    (True, dict(content="spam", no_encoding=None), True),
+    (True, dict(failed=True, msg="could not slurp"), False),
+])
+def test_copy_remote_file_to_dir(give_output_dir, result, expect_file, tmpdir):
```

Comment:
I suspect we could reuse whatever file transfer mechanism that Ansible has?

Reply:
We do... it uses the slurp module.

There is a "fetch" module which sounds like it would be useful but it does not put the file where I want and under the covers it is most likely to be calling slurp anyway. There is a 'synchronize' module which uses rsync if present (I don't want to rely on that on localhost). This is not at all as straightforward as I would expect :(

Reply:
Aha, haven't use that myself, unfortunate that is not easy to reuse.